akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -175,6 +179,43 @@ private[stream] object GraphInterpreter {
         outOwners.mkString("[", ",", "]") +
         ")"
   }
+
+  object GraphAssembly {
+    /**
+     * INTERNAL API
+     */
+    final def apply(inlets: immutable.Seq[Inlet[_]],
+                    outlets: immutable.Seq[Outlet[_]],
+                    stages: GraphStageWithMaterializedValue[_, _]*): GraphAssembly = {
+      // add the contents of an iterator to an array starting at idx
+      @tailrec def add[T](i: Iterator[T], a: Array[T], idx: Int): Array[T] =
```

Comment:
It's sad that arrays do not have methods to copy in stuff, so you wouldn't have to write these methods in the first place :S